macros/models/_project/yield_yak/yield_yak_deposits_withdraws.sql

```diff
@@ -0,0 +1,63 @@
+{%- macro yield_yak_deposits_withdraws(
+        blockchain = null,
+        event_name = 'Deposit'
+    )
+-%}
+
+{%- set namespace_blockchain = 'yield_yak_' + blockchain -%}
+
+WITH
+
+{% if is_incremental() -%}
+existing_contracts AS (
+    SELECT
+        t.contract_address
+        , MAX(t.block_number) AS latest_block_number
+    FROM
+    {{ this }} t
+    GROUP BY t.contract_address
+),
+{% endif -%}
+
+combined AS (
+    {%- for strategy in yield_yak_strategies(blockchain) %}
+        SELECT
+            s.contract_address
+            , s.evt_tx_hash AS tx_hash
+            , s.evt_index
+            , s.evt_block_time AS block_time
+            , s.evt_block_number AS block_number
+            , s.account AS user_address
+            , s.amount AS {{ event_name.lower() }}_amount
+        FROM {{ source(namespace_blockchain, strategy + '_evt_' + event_name) }} s
+        {%- if is_incremental() %}
+        LEFT JOIN
+        existing_contracts c
+            ON c.contract_address = s.contract_address
+        WHERE
+            ({{ incremental_predicate('s.evt_block_time') }}
+            AND s.evt_block_number > c.latest_block_number)
+            OR c.contract_address IS NULL -- This line allows for new contract_addresses being appended that were not already included in previous runs but also allows their entire historical data to be loaded
+        {%- endif %}
+        {% if not loop.last -%}
+        UNION ALL
+        {%- endif -%}
+    {%- endfor %}
+)
+
+SELECT
+    '{{ blockchain }}' AS blockchain
+    , c.contract_address
+    , c.tx_hash
+    , c.evt_index
+    , t.index AS tx_index
+    , c.block_time
+    , c.block_number
+    , c.user_address
+    , c.{{ event_name.lower() }}_amount
+FROM combined c
+LEFT JOIN
+{{ source(blockchain, 'transactions') }} t
+    ON t.hash = c.tx_hash
```

Comment:
Do you think we may add incremental filters on `transactions` too?

Reply:
Would be nice but I don't think that would work. There's a scenario where a Yield Yak contract has existed for a while but hasn't been decoded yet. If we then decoded that contract it will be picked up in the "combined" CTE (due to the `OR c.contract_address IS NULL` condition, but if we had incremental filters on `transactions` then the older transaction info for any such contract would not get picked up.

Reply:
even if we can't apply an incremental filter, we should be able to enhance the join.

can you add `block_number` to the join conditions?

transactions is also partitioned on `block_date` -- we could likely add condition for `block_date = cast(date_trunc('day', block_time) as date)`  (assuming `block_date` doesn't exist in the decoded tables. if it does, then simply join on `block_date`)

Reply:
also, can it be an inner join?
that's more efficient on the dunesql engine.

if inner join isn't possible, it's usually best to first read from the larger of the two tables, then left/right join as needed to smaller tables